src/weather/responses/weather.response.ts

```diff
@@ -0,0 +1,470 @@
+  chance_of_snow: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  vis_km: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  vis_miles: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  gust_mph: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  gust_kph: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  uv: number;
+}
+
+export class CurrentDto {
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  last_updated_epoch: number;
+
+  @ApiProperty({ type: String })
+  last_updated: string;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  temp_c: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  temp_f: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  is_day: number;
+
+  @ApiProperty({ type: ConditionDto })
+  condition: ConditionDto;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  wind_mph: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  wind_kph: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  wind_degree: number;
+
+  @ApiProperty({ type: String })
+  wind_dir: string;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  pressure_mb: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  pressure_in: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  precip_mm: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  precip_in: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  humidity: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  cloud: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  feelslike_c: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  feelslike_f: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  windchill_c: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  windchill_f: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  heatindex_c: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  heatindex_f: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  dewpoint_c: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  dewpoint_f: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  vis_km: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  vis_miles: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  uv: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  gust_mph: number;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  gust_kph: number;
+}
+
+export class ForecastDayDto {
+  @ApiProperty({ type: String })
+  date: string;
+
+  @ApiProperty({ type: Number })
+  @IsNumber()
+  date_epoch: number;
+
+  @ApiProperty({ type: DayForecastDto })
+  day: DayForecastDto;
+
+  @ApiProperty({ type: AstroDto })
+  astro: AstroDto;
+
+  @ApiProperty({ type: [HourDto] })
+  hour: HourDto[];
+}
+
+export class ForecastDto {
+  @ApiProperty({ type: [ForecastDayDto] })
+  forecastday: ForecastDayDto[];
+}
+
+export class WeatherApiResponseDto {
```

Comment:
I'd leave this DTO class and move others to separate files - otherwise this file is very big and looking at the beginning of the file you don't see what is the "main" class that we are exporting here